harvester/harvest/ogm.py

```diff
@@ -0,0 +1,351 @@
+        if not target_commit:
+            return []
+
+        # get all modified files SINCE this commit
+        changes = self._get_modified_files_since_commit(target_commit)
+
+        # parse modified file tuples into OGMRecord instances
+        for change_type, relative_filename in changes:
+            filename = os.path.join(self.local_repository_directory, relative_filename)
+            if change_type in ["A", "M", "C"]:
+                event = "created"
+            elif change_type in ["D"]:
+                event = "deleted"
+            else:
+                message = f"Git file change type not handled: '{change_type}'"
+                logger.error(message)
+                continue
+            yield OGMRecord(
+                identifier=self.create_identifier_from_filename(filename),
+                filename=filename,
+                relative_filename=relative_filename,
+                harvest_event=event,
+                git_change_type=change_type,
+                target_commit=target_commit,
+            )
+
+    def _get_commit_before_date(self, target_date: str) -> Commit | None:
+        """Identify last commit BEFORE a target date."""
+        # raise exception if date is before epoch time of 1979-01-01
+        if date_parser(target_date) < date_parser("1979-01-01"):
+            raise OGMFromDateExceedsEpochDateError
+
+        from_timestamp = date_parser(target_date).timestamp()
+        local_repo = self.clone_repository()
+        commits = list(local_repo.iter_commits(since=from_timestamp))
+        if not commits:
+            message = f"Could not find any commits after date: {target_date}"
+            logger.info(message)
+            return None
+        earliest_commit = commits[-1]
+        if not earliest_commit.parents:
+            target_commit = earliest_commit
+        else:
+            target_commit = commits[-1].parents[0]
+        message = (
+            f"Last commit before date '{target_date}': "
+            f"{target_commit.committed_datetime.isoformat()}, {target_commit.hexsha}"
+        )
+        logger.debug(message)
+        return target_commit
+
+    def _get_modified_files_since_commit(self, target_commit: Commit) -> list[list[str]]:
+        """Get all modified files SINCE a target commit."""
+        app_working_dir = os.getcwd()
+        os.chdir(self.local_repository_directory)
+        cmd_args = [
+            "git",
+            "diff",
+            "--no-renames",
+            "--name-status",
+            f"{target_commit.hexsha}..HEAD",
+        ]
+        logger.debug(cmd_args)
+        output = subprocess.check_output(cmd_args).decode()  # noqa: S603
+        os.chdir(app_working_dir)
+        return [change.split("\t") for change in output.split("\n") if change != ""]
+
+    def filter_records(self, records: Iterator["OGMRecord"]) -> Iterator["OGMRecord"]:
+        """Filter files based on repository config."""
+        if "filename_regex" in self.config:
```

Comment:
Do we have to account for both `filename_regex` and  `file_directory` possibly being present?

Reply:
Another good suggestion.  I had considered if a repo's configuration should allow multiple strategies, or if there might be more strategies, but it seems like these two easily cover our scoped OGM repos.  

This prompted a little refactor into two methods:

1. `_get_filter_regex()` - parse config and return a regex expression or raise errors (like yours above)
2. `filter_records()` - very short and simple, apply the filter and yield records